odd-platform-api/src/main/java/org/opendatadiscovery/oddplatform/repository/reactive/ReactiveEnumValueRepositoryImpl.java

```diff
@@ -0,0 +1,65 @@
+package org.opendatadiscovery.oddplatform.repository.reactive;
+
+import java.util.List;
+import java.util.Map;
+import org.jooq.Condition;
+import org.jooq.Field;
+import org.jooq.UpdateResultStep;
+import org.jooq.impl.DSL;
+import org.opendatadiscovery.oddplatform.model.tables.pojos.EnumValuePojo;
+import org.opendatadiscovery.oddplatform.model.tables.records.EnumValueRecord;
+import org.opendatadiscovery.oddplatform.repository.util.JooqQueryHelper;
+import org.opendatadiscovery.oddplatform.repository.util.JooqReactiveOperations;
+import org.opendatadiscovery.oddplatform.repository.util.JooqRecordHelper;
+import org.springframework.stereotype.Repository;
+import reactor.core.publisher.Flux;
+
+import static org.opendatadiscovery.oddplatform.model.Tables.ENUM_VALUE;
+
+@Repository
+public class ReactiveEnumValueRepositoryImpl
+    extends ReactiveAbstractSoftDeleteCRUDRepository<EnumValueRecord, EnumValuePojo>
+    implements ReactiveEnumValueRepository {
+    private final JooqRecordHelper jooqRecordHelper;
+
+    public ReactiveEnumValueRepositoryImpl(final JooqReactiveOperations jooqReactiveOperations,
+                                           final JooqRecordHelper jooqRecordHelper,
+                                           final JooqQueryHelper jooqQueryHelper) {
+        super(jooqReactiveOperations, jooqQueryHelper, ENUM_VALUE, EnumValuePojo.class);
+        this.jooqRecordHelper = jooqRecordHelper;
+    }
+
+    @Override
+    public Flux<EnumValuePojo> getEnumValuesByFieldId(final long datasetFieldId) {
+        final var conditionWithSoftDeleteFilter = addSoftDeleteFilter(
+            ENUM_VALUE.DATASET_FIELD_ID.eq(datasetFieldId));
+
+        final var query = DSL.selectFrom(ENUM_VALUE)
+            .where(conditionWithSoftDeleteFilter);
+
+        return jooqReactiveOperations.flux(query)
+            .map(r -> r.into(EnumValuePojo.class));
+    }
+
+    public Flux<EnumValuePojo> softDeleteOutdatedEnumValues(final long datasetFieldId,
+                                                            final List<Long> idsToKeep) {
+        if (idsToKeep.isEmpty()) {
```

Comment:
I'm not sure, that this is expected behaviour. If it's empty we need to delete ALL enum values

Reply:
Thanks for catching it! This is definitely incorrect behavior, I'll remove this check